cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/internal/AliasConfiguration.qll

```diff
@@ -95,3 +98,31 @@ class IndirectParameterAllocation extends Allocation, TIndirectParameterAllocati
 
   final override predicate alwaysEscapes() { none() }
 }
+
+class DynamicAllocation extends Allocation, TDynamicAllocation {
+  CallInstruction call;
+
+  DynamicAllocation() { this = TDynamicAllocation(call) }
+
+  final override string toString() {
+    result = call.toString() + " at " + call.getLocation() // TODO: make this both short and unique
```

Comment:
Are these displayed outside of test code? I didn't come up with a good way to handle this, but if it's not visible in real results I'm fine with this,

Reply:
At the moment, I don't believe that `Allocation` is part of the public-facing API, so that `toString()` should only be used in test/dump code. I think it's fine as it is, but I'd probably leave some form of that comment in the code.